File: spoj py/PTIT126E.py
s=raw_input()
while s[0]!='#':
    Y=0
    N=0
    A=0
    for k in s:
        if k=='Y':
            Y=Y+1
        if k=='N':
            N=N+1
        if k=='A':
            A=A+1
    if A>=int(round(float(len(s))/2)):
        print 'need quorum'
    else:
        if Y>N:
            print 'yes'
        elif Y<N:
            print 'no'
        else:
            print 'tie'
    s=raw_input()

File: PTIT126E.cpp
#include<stdio.h>
#include<string.h>
#include<math.h>

int main(void)
{
	char str[100];
	scanf("%s",str);
	while(str[0]!='#')
	{
		int Y=0,N=0,A=0,n=strlen(str);
		for(int i=0;i<n;i++)
		{
			if(str[i]=='A')
				A++;
			if(str[i]=='Y')
				Y++;
			if(str[i]=='N')
				N++;
		}
		if(A>=ceil((float)n/2))
			printf("need quorum\n");
		else
		{
			if(Y>N)
				printf("yes\n");
			else if(Y<N)
				printf("no\n");
			else
				printf("tie\n");
		}
		scanf("%s",str);
	}
 return 0;
}

